src/share/util/shr_dmodel_mod.F90

```diff
@@ -872,6 +909,232 @@ subroutine shr_dmodel_readstrm(stream, pio_subsystem, pio_iotype, pio_iodesc, gs
+  if (present(istr)) then
+     lstr = trim(istr)
+  endif
+
+  bstr = ''
+  if (present(boundstr)) then
+     bstr = trim(boundstr)
+  endif
+
+  call t_barrierf(trim(lstr)//'_BARRIER',mpicom)
+  call t_startf(trim(lstr)//'_setup')
+  call MPI_COMM_RANK(mpicom,my_task,ierr)
+  master_task = 0
+
+  gsize = mct_gsmap_gsize(gsMap)
+  lsize = mct_gsmap_lsize(gsMap,mpicom)
+
+  if (my_task == master_task) then
+     localCopy = (shr_file_queryPrefix(path) /= shr_file_noPrefix )
+     if (localCopy) then
+        call shr_file_get(rCode,fn,trim(path)//fn)
+        fileName = fn
+     else                 ! DON'T acquire a local copy of the data file
+        fileName = trim(path)//fn
+     end if
+     inquire(file=trim(fileName),exist=fileExists)
+     if (.not. fileExists) then
+        write(logunit,F00) "ERROR: file does not exist: ", trim(fileName)
+        call shr_sys_abort(subName//"ERROR: file does not exist: "//trim(fileName))
+     end if
+  endif
+
+  if (my_task == master_task) then
+     call shr_stream_getFileFieldName(stream,1,sfldName)
+  endif
+
+  call t_stopf(trim(lstr)//'_setup')
+
+  if (pio_iotype == iotype_std_netcdf) then
+
+     write(logunit,F01) "shr_dmodel_readstrm_fullfile: not supported for iotype_std_netcdf"
+     call shr_sys_abort(subname//"ERROR extend shr_dmodel_readstrm_fullfile")
+
+  else
+
+     call t_startf(trim(lstr)//'_readpio')
+     call shr_mpi_bcast(sfldName,mpicom,'sfldName')
+     call shr_mpi_bcast(filename,mpicom,'filename')
+
+     call shr_stream_getCurrFile(stream,fileopen=fileopen,currfile=currfile,currpioid=pioid)
+
+     if (fileopen .and. currfile==filename) then
+        ! don't reopen file, all good
+     else
+        ! otherwise close the old file if open and open new file
+        if (fileopen) then
+           if (my_task == master_task) then
+              write(logunit,F00) 'close  : ',trim(currfile)
+              call shr_sys_flush(logunit)
+           endif
+           call pio_closefile(pioid)
+        endif
+        if (my_task == master_task) then
+           write(logunit,F00) 'open   : ',trim(filename)
+           call shr_sys_flush(logunit)
+        endif
+        rcode = pio_openfile(pio_subsystem, pioid, pio_iotype, trim(filename), pio_nowrite)
+        call shr_stream_setCurrFile(stream,fileopen=.true.,currfile=trim(filename),currpioid=pioid)
+
+        call pio_seterrorhandling(pioid,PIO_INTERNAL_ERROR)
+
+        rcode = pio_inq_varid(pioid,trim(sfldName),varid)
+        rcode = pio_inq_varndims(pioid, varid, ndims)
+        allocate(dimid(ndims))
+
+        rcode = pio_inq_vardimid(pioid, varid, dimid(1:ndims))
+
+        nx = 1
+        ny = 1
+        nz = 1
+        if (ndims >= 1) rcode = pio_inq_dimlen(pioid, dimid(1), nx)
+        if (ndims >= 2) rcode = pio_inq_dimlen(pioid, dimid(2), ny)
+        if (ndims >= 3) rcode = pio_inq_dimlen(pioid, dimid(3), nz)
+        deallocate(dimid)
+
+        if (gsize /= nx*ny .and. gsize /= nx*ny*nz) then
```

Comment:
Just realized this conditional should be modified to `if (gsize /= nx*ny) ` because nz is assumed to ntimes.